src/gamestates/gameState.lua

```diff
@@ -1,43 +1,15 @@
-local GameState = {}
-GameState.__index = GameState
+local enabledFunctions = require("gamestates/enabledFunctions")
 
-function GameState.load()
-end
-
-function GameState.update()
-end
-
-function GameState.draw()
-end
-
-function GameState.keypressed()
-end
-
-function GameState.keyreleased()
-end
-
-function GameState.mousepressed()
-end
-
-function GameState.mousereleased()
-end
-
-function GameState.joystickpressed()
-end
+local function emptyFunction() end
 
-function GameState.joystickreleased()
+local function index(t, key)
```

Comment:
If you wanted, you could call this function `__index` and then below you could do `setmetatable({__index = __index})`. You could of course also do this with `call`. I'm not sure if that's better or not. I guess it's up to you whether you want to free up the `index` and `call` names to be used for other things, like what you were saying about the functions in Class.

Reply:
As soon as you do {__index = index} the name of the original function is irrelevant. I don't plan on adding anything with those names in that file ever, so I am fine with those place holders in that file.